src/material/input/input.ts

```diff
@@ -308,7 +310,7 @@ export class MatInput
     // exists on iOS, we only bother to install the listener on iOS.
     if (_platform.IOS) {
       ngZone.runOutsideAngular(() => {
-        _elementRef.nativeElement.addEventListener('keyup', (event: Event) => {
+        this._iOSKeyupListener = (event: Event) => {
```

Comment:
Rather than assigning the listener conditionally, it should be an arrow function on the class itself.